File: scala/scalatra_for_dummies/src/main/scala/ExtremeStartup.scala
import org.scalatra.ScalatraServlet


class ExtremeStartup extends ScalatraServlet {

    def answer(question:String):String = {
        val PlusExpression = """.*what is the sum of (\d+) and (\d+)""".r
        val SantaClauseExpression = """.*what's santa clause's real name""".r
        question match {
            case PlusExpression(x,y) =>
                return (Integer.parseInt(x) + Integer.parseInt(y)).toString
            case SantaClauseExpression() => // <-- Parens are important here!
                return "Kris Kringle"
            case _ => 
                System.out.println("Unrecognized question: " + question)
                return "scala team"
        }
    }

    get("/") {
      params("q") match {
        case q:String => 
            System.out.println("A request has arrived")
            answer(q)
        case _ => pass()
      }
    }
    
}

File: ruby/sinatra_for_dummies/extreme_startup.rb
require 'rubygems'
require 'sinatra'

class ExtremeStartup

  def answer(q)
    if q =~ /the sum of (\d+) and (\d+)/
      return ($1.to_i + $2.to_i).to_s 
    elsif q =~ /what's santa clause's real name/
      return "Kris Kringle"
    else
      return "team name"
    end
  end

end

server = ExtremeStartup.new

configure do
  set :bind, '0.0.0.0'
  set :port, 1337
end

get '/' do
  puts "A request has arrived"
  server.answer(params[:q])
end
